R/template.R

```diff
@@ -468,34 +594,55 @@ curve_survival_metric_summarizer <- function(name,
     error_call = error_call
   )
 
-  if (!quo_is_null(case_weights)) {
+  if (quo_is_null(case_weights)) {
+    group_case_weights <- NULL
+  } else {
     case_weights <- yardstick_eval_select(
       expr = case_weights,
       data = data,
       arg = "case_weights",
       error_call = error_call
     )
-
-    case_weights <- expr(.data[[!!case_weights]])
   }
 
-  out <- dplyr::reframe(
-    data,
-    .metric = .env[["name"]],
-    .estimator = finalize_estimator(
-      .data[[truth]],
-      metric_class = .env[["name"]]
-    ),
-    .estimate = fn(
-      truth = .data[[truth]],
-      estimate = .data[[estimate]],
-      case_weights = !!case_weights,
-      na_rm = .env[["na_rm"]],
-      !!!fn_options
+  group_rows <- dplyr::group_rows(data)
+  groups <- vctrs::vec_chop(data, indices = group_rows)
+  out <- vector("list", length = length(groups))
+
+  for (i in seq_along(groups)) {
+    group <- groups[[i]]
+
+    group_truth <- group[[truth]]
+    group_estimate <- prob_estimate_convert(group[estimate])
+
+    if (is_string(case_weights)) {
+      group_case_weights <- group[[case_weights]]
+    }
+
+    elt_out <- c(
+      dplyr::group_keys(group),
+      list(
+        .metric = name,
+        .estimator = finalize_estimator(
+          group_truth,
+          metric_class = name
+        ),
+        .estimate = rlang::inject(
+          fn(
+            truth = group_truth,
+            estimate = group_estimate,
+            case_weights = group_case_weights,
+            na_rm = na_rm,
+            !!! fn_options
+          )
+        )
+      )
     )
-  )
 
-  dplyr::as_tibble(out)
+    out[[i]] <- tibble::as_tibble(elt_out)
```

Comment:
```suggestion
    out[[i]] <- tibble::new_tibble(elt_out)
```

Reply:
dealt with in [tidymodels/yardstick@`0c17e1b` (#405)](https://github.com/tidymodels/yardstick/pull/405/commits/0c17e1bbee9c66272c3956b95897ccb5c984be7f)